cmd/lncli/commands.go

```diff
@@ -728,28 +756,216 @@ func closeChannel(ctx *cli.Context) error {
+			"channels will be closed unilaterally, so funds " +
+			"within them will be locked for a few blocks (CSV " +
+			"delay) before they can be spent. (yes/no): "
+
+		confirmed := promptForConfirmation(msg)
+
+		// We can safely exit if the user did not confirm.
+		if !confirmed {
+			return nil
+		}
+	default:
+		for i, channel := range channelsToClose {
+			// If the channel is inactive, we will attempt to force
+			// close the channel and prompt the user for
+			// confirmation.
+			if !channel.GetActive() {
+				msg := fmt.Sprintf("Unilaterally close channel "+
+					"with node %s? The closing transaction "+
+					"will need %d confirmations before "+
+					"the funds can be spent. (yes/no): ",
+					channel.RemotePubkey, channel.CsvDelay)
+
+				confirmed := promptForConfirmation(msg)
+
+				// If the user did not confirm the force close,
+				// remove the channel from our list.
+				if !confirmed {
+					channelsToClose = append(channelsToClose[:i],
+						channelsToClose[i+1:]...)
+				}
+			} else if ctx.Bool("inactive_only") {
+				// Otherwise, we'll remove it from our list if
+				// we were requested to close inactive channels
+				// only.
+				channelsToClose = append(channelsToClose[:i],
+					channelsToClose[i+1:]...)
+			}
+		}
+	}
+
+	// result defines the result of closing a channel. The closing
+	// transaction ID is populated if a channel is successfully closed.
+	// Otherwise, the error that prevented closing the channel is populated.
+	type result struct {
+		RemotePubKey string `json:"remote_pub_key"`
+		ChannelPoint string `json:"channel_point"`
+		ClosingTxid  string `json:"closing_txid"`
+		FailErr      string `json:"error"`
+	}
+
+	// Launch each channel closure in a goroutine in order to execute them
+	// in parallel. Once they're all executed, we will print the results as
+	// they come.
+	resultChan := make(chan result, len(channelsToClose))
+	for _, channel := range channelsToClose {
+		go func(channel *lnrpc.ActiveChannel) {
+			res := result{}
+			res.RemotePubKey = channel.RemotePubkey
+			res.ChannelPoint = channel.ChannelPoint
+			defer func() { resultChan <- res }()
+
+			// Parse the channel point in order to create the close
+			// channel request.
+			s := strings.Split(res.ChannelPoint, ":")
+			if len(s) != 2 {
+				res.FailErr = "expected channel point with " +
+					"format txid:index"
+				return
+			}
+			index, err := strconv.ParseUint(s[1], 10, 32)
 			if err != nil {
-				return err
+				res.FailErr = fmt.Sprintf("unable to parse "+
+					"channel point output index: %v", err)
+				return
 			}
 
-			printJSON(struct {
-				ClosingTXID string `json:"closing_txid"`
-			}{
-				ClosingTXID: txid.String(),
-			})
+			req := &lnrpc.CloseChannelRequest{
+				ChannelPoint: &lnrpc.ChannelPoint{
+					FundingTxid: &lnrpc.ChannelPoint_FundingTxidStr{
+						FundingTxidStr: s[0],
+					},
+					OutputIndex: uint32(index),
+				},
+				Force: !channel.GetActive(),
```

Comment:
You can set `block: false` for readability

Reply:
`block` is actually not used in the `lnrpc.CloseChannelRequest`, it's just a flag used by the `closechannel` command.

Reply:
Ah 😅